bottlecap/src/lifecycle/invocation/triggers/sqs_event.rs

```diff
@@ -196,6 +196,81 @@ impl Trigger for SqsRecord {
         // TODO: AWSTraceHeader
         carrier
     }
+
+    fn is_async(&self) -> bool {
+        true
+    }
+}
+
+// extractTraceContextfromAWSTraceHeader extracts trace context from the
+// AWSTraceHeader directly. Unlike the other carriers in this file, it should
+// not be passed to the tracer.Propagator, instead extracting context directly.
+pub(crate) fn extract_trace_context_from_aws_trace_header(
```

Comment:
Why a crate method and not just a method for the struct? We are not using it anywhere else

Reply:
I prefer the flexibility and not adding extra restrictions that aren't needed. It makes testing and changing code simpler.
 I also think a lot of the structs and traits should go away, so would rather avoid adding code and complexity to keep doing that